### PR TITLE
HTS precompile `transferFrom` support

### LINKED ISSUE #3904: Add ERC transferFrom support in IHederaTokenService precompile contract
### Problem

Currently the `IERC20(token).transferFrom(sender, recipient, amount)` and `IERC721(token).transferFrom(from, to, tokenId)` functions are only accessible via redirects.
However, this makes for a sub optimal developer experience.
More importantly this was a gap noted in [HIP 376](https://hips.hedera.com/hip/hip-376)

### Solution

Add the following functions as precompile methods
- `transferFrom(address sender, address recipient, uint256 amount) external returns (bool)` - Works by creating a synthetic CryptoTransferTransaction with fungible token transfers with the is_approval property set to true.
- `transferFrom(address from, address to, uint256 tokenId) external payable` - Works by creating a synthetic CryptoTransferTransaction with nft token transfers with the is_approval property set to true.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by dimitar-dinev**
**Description**:

This PR is a continuation of #3638, adding `IHederaTokenService` support of the ERC20/721 `transferFrom` method.

**Related issue(s)**:

Fixes #3904

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java [hunk 1]
```diff
@@ -154,9 +197,15 @@ public void run(final MessageFrame frame) {
         }
 
         if (isFungible) {
-            frame.addLog(getLogForFungibleTransfer(asTypedEvmAddress(tokenID)));
+            frame.addLog(
+                    getLogForFungibleTransfer(
+                            asTypedEvmAddress(
+                                    tokenID != null ? tokenID : extractTokenIdFromSyntheticOp())));
```
Comment (shemnon) [inline hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java:203]: Do this `asTypedEvmAddress` calculation and the one at line 207-298 once at line 198.

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java [hunk 2]
```diff
@@ -740,6 +740,38 @@ yield switch (nestedFunctionSelector) {
                             syntheticTxnFactory,
                             infrastructureFactory,
                             precompilePricingUtils);
+                    case AbiConstants.ABI_ID_TRANSFER_FROM -> checkFeatureFlag(
+                            dynamicProperties.areAllowancesEnabled(),
+                            () ->
+                                    new ERCTransferPrecompile(
```
Comment (shemnon) [inline hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java:746]: I wonder if we should look into a HTSPrecompiledContract local Dagger Module that can produce these objects.  There are a lot of constructor params that are just passed directly from the constructor to the instance.

possibly https://dagger.dev/dev-guide/assisted-injection

Not for this PR.

**Review by stoqnkpL — APPROVED** [review]
LGTM

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java (hunk 1 above)
Comment (dimitar-dinev) [reply hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java]: Fixed

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**